docs/protocol.md

```diff
@@ -1062,6 +1063,118 @@ RPC
 ```
 </details>
 
+### Catching up on past messages on a channel using paging
+
+🧭 **RPC Message** > **Query** > **Paged Catchup**
+
+By executing a paged catchup action, a client can ask the server to receive a specified number of
+past messages on a specific channel.
+
+For now, it is to be used to retrieve chirps on the social media channel by paging when a new client joins the LAO instead of getting all chirps at once. This is done in an effort to reduce network traffic at catchup.
+
+This may serve as a starting point for the paging of messages in other channels as a future optimization.
```

Comment:
I would try to clarify this further:
What does past messages mean here? (i.e. what is the order)
Why was the message designed this way? (basically what we discussed, this gives some background information)

What is "the social media channel"? I would try to be more precise here.
Also the docs say it is only used for chirps, what should happen if this is sent on some other channel? Is it a no-op for now? Or should the server try to find chirps on any channel and return them?
And which chirp message(-s) should this return? (I understand but the documentation should be precise in this regard)

Reply:
Noted
As for the "Also the docs say it is only used for chirps, what should happen if this is sent on some other channel? Is it a no-op for now? Or should the server try to find chirps on any channel and return them?" question, we decided to not implement paging for other channels this semester.

Reply:
I assumed so but the message may be sent by (malicious) actors anyway. The docs should describe how to process the message even if it is sent in an unexpected way.

Reply:
I think @Tyratox here makes a very important point.

> we decided to not implement paging for other channels this semester.

That's perfectly okay, but what is the expected behavior when it *does* get called on a non-chirp channel ? Not leaving this kind of ambiguity in the specification is good for two reasons.

1. Unexpected stuff happens. Even if there are no malicious actors, a front-end could have bugs and make the wrong RPC call. What happens *then* ?
2. Without a clear specification, I can guarantee that different backends will spontaneously implement this (slightly) differently. It doesn't cost much to clarify and say "we return an error" or "we default to the normal catchup behavior". 
3. Conversely, if there's a bug and the points above are not addressed, you'll have to debug behaviors like "in some cases, with fe2 and be1 [for instance], the loading of chirps fails with some weird error". 

As such:
- if you pick to revert to a reasonable "default" behavior, such as behaving more or less like a catchup (without paging), document that !
- if you pick to return an error, define the error (what code ? what message ?), try and make it super unambiguous and consistent ! 

Best of luck :-)

Reply:
Noted
Thanks for the feedback